raiden/messages/metadata.py

```diff
@@ -101,18 +103,23 @@ def _post_load(  # pylint: disable=no-self-use,unused-argument
         return data
 
     @post_dump(pass_many=True)
-    def _post_dump(  # pylint: disable=no-self-use,unused-argument
+    def _post_dump(  # pylint: disable=unused-argument
         self, data: Dict[str, Any], many: bool
     ) -> Dict[str, Any]:
-        original_data = data.pop("original_data", {})
-        if original_data:
-            # We received the metadata (Mediator/Target) and read in the data
-            # for internal processing,
-            # so once we pass them to the next node just dump the data exactly as we received them
-            return original_data
-        # We initially created the Metadata (Initiator),
-        # so dump the known fields as per the Schema
-        return data
+        """
+        `original_data` means we received the metadata (Mediator/Target) and read in the data
+        for internal processing, so once we pass them to the next node just dump the data exactly
+        as we received them.
+
+        Returning `data` means we initially created the Metadata (Initiator), so dump the known
+        fields as per the Schema
+        """
+        dumped_data = data.pop("original_data", None) or data
+
+        if not self.Meta.serialize_missing:
```

Comment:
I found out what's the problem you had:
The `post_dump` method from the BaseSchema is NOT used for the highest level Schema upon serialisation, but only for schemas of nested attributes on this Schema.(I don't know if this is expected or a bug on our side?)

But that's why the mechanism is working for the RouteMetadata that is part of the Metadata object, but not for attributes directly associated on the Metadata. 
So my assumption was wrong that simply adding the `serialize_missing` `class Meta`-attribute would suffice, although it was intended to be this way.
I can elaborate in private if that doesn't make sense initially - but I'd rather fix the issue if possible instead of implementing it twice

Reply:
✅